yt_dlp/extractor/tiktok.py

```diff
@@ -344,76 +349,76 @@ def extract_addr(addr, add_meta={}):
             '_format_sort_fields': ('quality', 'codec', 'size', 'br'),
         }
 
-    def _parse_aweme_video_web(self, aweme_detail, webpage_url):
+    def _parse_aweme_video_web(self, aweme_detail, webpage_url, video_id):
         video_info = aweme_detail['video']
         author_info = traverse_obj(aweme_detail, 'authorInfo', 'author', expected_type=dict, default={})
         music_info = aweme_detail.get('music') or {}
         stats_info = aweme_detail.get('stats') or {}
-        user_url = self._UPLOADER_URL_FORMAT % (traverse_obj(author_info,
-                                                             'secUid', 'id', 'uid', 'uniqueId',
-                                                             expected_type=str_or_none, get_all=False)
-                                                or aweme_detail.get('authorSecId'))
+        channel_id = traverse_obj(author_info or aweme_detail, (('authorSecId', 'secUid'), {str}), get_all=False)
+        user_url = self._UPLOADER_URL_FORMAT % channel_id if channel_id else None
 
         formats = []
-        play_url = video_info.get('playAddr')
-        width = video_info.get('width')
-        height = video_info.get('height')
-        if isinstance(play_url, str):
-            formats = [{
+        width = int_or_none(video_info.get('width'))
+        height = int_or_none(video_info.get('height'))
+
+        for play_url in traverse_obj(video_info, ('playAddr', (({list}, ..., 'src'), ({str})), {url_or_none})):
```

Comment:
`({str})` -  the brackets are useless. Also, it's not needed due to `url_or_none` later
`{list}, ...` - what's the `{list}` for?

Reply:
`{list}` and `{str}` are used to branch depending on type, as `playAddr` can be a list or a string

Reply:
```suggestion
        for play_url in traverse_obj(video_info, ('playAddr', ((..., 'src'), None), {url_or_none})):
```

Is this what u want?

Reply:
yep, ig i overthought that one